src/extensions/scratch3_translate/index.js

```diff
@@ -247,6 +247,10 @@ class Scratch3TranslateBlocks {
      * @return {Promise} - a promise that resolves after the response from the translate server.
      */
     getTranslate (args) {
+        // If the text contains only digits 0-9 and nothing else, return it without
+        // making a request.
+        if (/^\d+$/.test(args.WORDS)) return Promise.resolve(args.WORDS);
```

Comment:
Do we need the promise.resolve? Below (line ~253) we don't have one, though I expect it doesn't matter, but asking because I've forgotten how this fits in with the block execution :-/

Reply:
I don't think it matters too much - returing just args could break projects that rely on Promise block execution inside warp mode procedures but in general you shouldn't use these blocks inside warp procedure anyway.

Reply:
Yeah, looks like this will be okay, since we were already inconsistent. In normal use, where the API returns a result, the opcode returns a promise. If we hit the local cache for the same exact request, we return the value. This change adds returning a promise for the numbers-only case.